src/features/providers/components/provider-dialog.tsx

```diff
@@ -0,0 +1,42 @@
+import {
+  DialogModalOverlay,
+  DialogModal,
+  Dialog,
+  DialogHeader,
+  DialogTitle,
+  DialogCloseButton,
+} from "@stacklok/ui-kit";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+
+export function ProviderDialog({
+  title,
+  children,
+}: {
+  title: string;
+  children: React.ReactNode;
+}) {
+  const [isModalOpen, setIsModalOpen] = useState(true);
+  const navigate = useNavigate();
+
+  return (
+    <DialogModalOverlay
+      isDismissable={false}
+      isOpen={isModalOpen}
+      onOpenChange={(isOpen) => {
+        setIsModalOpen(isOpen);
+        navigate("/providers");
+      }}
+    >
+      <DialogModal>
+        <Dialog width="md" className="">
+          <DialogHeader>
+            <DialogTitle>{title}</DialogTitle>
+            <DialogCloseButton />
+          </DialogHeader>
+          {children}
```

Comment:
```suggestion
        <DialogContent>
          {children}
        </DialogContent>
```

Adding this will handle overflow with a scrollbar

**Before**

<img width="827" alt="Screenshot 2025-02-05 at 10 58 50 AM" src="https://github.com/user-attachments/assets/9223f2f2-82cb-48f0-a5bd-d3bc28478610" />

**After**

<img width="982" alt="Screenshot 2025-02-05 at 10 59 20 AM" src="https://github.com/user-attachments/assets/a450e94e-d2df-417e-90c4-63b1fadf5ac8" />

Reply:
the `DialogContent` is inside the children

Reply:
Ah, I've just spotted that this is already being handled in the routes that contain the forms:

```tsx
    <ProviderDialog title="Create Provider">
      <Form onSubmit={handleSubmit} validationBehavior="aria">
        <DialogContent className="p-8">
          <ProviderForm provider={provider} setProvider={setProvider} />
        </DialogContent>
        <ProviderDialogFooter />
      </Form>
    </ProviderDialog>
```

Ignore my previous suggestion, I think we need to wrap the form around the entire Dialog like this:

```
    <Form onSubmit={handleSubmit} validationBehavior="aria">
      <ProviderDialog title="Create Provider">
        <DialogContent>
          <ProviderForm provider={provider} setProvider={setProvider} />
        </DialogContent>
        <ProviderDialogFooter />
      </ProviderDialog>
    </Form>
    
```

Reply:
mm I guess that if we move the form above the `DialogModalOverlay` we will have an issue triggering the submit and the modal, I experimented this already 🤔 .
What is the value to move the Form up wrapping the entire modal?

Reply:
the form component is breaking the styling interaction of the Dialog components — DialogContent already handles overflow with scrolling, but not in it's current form.

I'm suggesting to wrap the entire dialog, so you can still use the Button with `type="submit"` in the footer — but there's many ways to slice this

Reply:
fixed [here](https://github.com/stacklok/codegate-ui/pull/253/commits/85e769be586b43c83a206386658017e75b5f6d24)